backend/prompt_studio/prompt_studio_core/prompt_studio_helper.py

```diff
@@ -702,7 +702,7 @@ def dynamic_indexer(
                 chunk_overlap=profile_manager.chunk_overlap,
                 reindex=reindex,
                 output_file_path=extract_file_path,
-                usage_kwargs=usage_kwargs,
+                usage_kwargs=usage_kwargs.copy(),
```

Comment:
@gaya3-zipstack why is this copy needed?

Reply:
This is to have call by value and not reference to avoid any modification to the variable before being used in other places. In SDK, we add additional params like adapter_instance_id to usage_kwargs. If we use the same usage_kwargs to initialise first, the LLM and then Embedding, when pass the variable for the second time (for Embedding), it will have the value of the LLM also. To avoid this, copy is used here.